### PR TITLE
Standardize tab navigability in web apps forms

### PR DESCRIPTION
**Pull request by nospame**
## Product Description
In line with various web accessibility guidance, only interactive elements will be "focusable" in web apps forms. This means, when navigating with "tab", elements that are selected have some user interaction like an input, button, or link.

## Technical Summary
[USH-4106](https://dimagi.atlassian.net/browse/USH-4106) and notably discussion in its comments. Removes `tabindex` from headings and label text and removes focus for empty `<audio>` and `<video>` elements. Also some small cleanup like removing a `for` attribute from a `<span>` (`for` is only relevant on a `<label>`) and replacing an italic `<i>` with a styled `<span>`.

### Safety story
These changes affect accessibility only and are generally safe _for most users_. Have tested locally with a screen reader and clicking through accessibility tree to see that inputs are correctly labeled.

### QA Plan
Regression is unlikely, but this should go through a QA cycle for accessibility, to ensure all inputs are properly labeled by their labels and that all interactive elements _are_ focusable.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-4106]: https://dimagi.atlassian.net/browse/USH-4106?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/cloudcare/templates/form_entry/templates.html [hunk 1]
```diff
@@ -266,10 +265,10 @@ <h1 class="title" data-bind="text: title, visible: !showInFormNavigation()"></h1
         </div>
         <!-- /ko -->
       </div>
-      <i class="hint-text" data-bind="
+      <em class="hint-text" data-bind="
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/templates/form_entry/templates.html:268]: Why emphasis on a hint text?

**Review by MartinRiese — APPROVED** [review]

File: corehq/apps/cloudcare/templates/form_entry/templates.html (hunk 1 above)
Comment (nospame) [reply corehq/apps/cloudcare/templates/form_entry/templates.html:268]: We use `<i>` pretty widely to mean "icon" and I found it slightly confusing. But you're probably right that "emphasized" isn't correct for simply italic hint text, either. I don't have incredibly strong feelings, does it seem better to leave it as `<i>`?

Comment (MartinRiese) [reply corehq/apps/cloudcare/templates/form_entry/templates.html:268]: I was more wondering if a screen reader uses it somehow. if now. I am fine either way

Comment (nospame) [reply corehq/apps/cloudcare/templates/form_entry/templates.html]: It looks like using either `i` or `em` moves it up one level in the accessibility tree, versus using `span` which keeps it at the same level as the label text, so I opted for the latter. https://github.com/dimagi/commcare-hq/pull/34084/commits/45661a5725ad71f95f7ea5da2c1e353a43eb1298

**Comment by nospame** [comment]
Requesting review/re-review here please. Thank you!

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js [hunk 2]
```diff
@@ -1343,14 +1344,27 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
...
 
+        applySelect2Labelledby(entry.entryId, isMulti);
+    }
+
+    /**
+     * Select2 ignores labelling from original `<select>` element.
+     * Fix that by applying `aria-labelledby` to the element getting keyboard focus.
+     */
+    function applySelect2Labelledby(entryId, isMulti) {
+        var $input = $('#' + entryId);
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:1363]: I think it is fair to use `const` and `let` in new code.

Comment (nospame) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js]: Seems fair to me. https://github.com/dimagi/commcare-hq/pull/34084/commits/3b28124c76f3448b18b59c060bfc3e5561a4593b

File: corehq/apps/cloudcare/templates/form_entry/templates.html [hunk 3]
```diff
@@ -56,18 +56,12 @@
...
           <span class="webapp-markdown-output caption" data-bind="html: caption_markdown() || caption(), attr: {id: captionId()}"></span>
           <i class="fa fa-warning text-danger pull-right" data-bind="visible: hasError() && !showChildren()"></i>
-          <button class="btn btn-danger del pull-right" href="#" data-bind="
+          <button class="btn btn-danger del pull-right" href="#" aria-label="{% trans_html_attr 'Delete Repeat' %}"
+                  data-bind="
                       visible: isRepetition,
                       click: deleteRepeat
                   "><i class="fa fa-remove"></i></button>
         </div>
-        <span class="webapp-markdown-output caption" data-bind="html: caption_markdown() || caption(), attr: {id: captionId()}"></span>
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/templates/form_entry/templates.html]: Is this just duplicate code? Looks very similar to the span and button above.

Comment (MartinRiese) [reply corehq/apps/cloudcare/templates/form_entry/templates.html]: thanks for looking into it.

**Review by MartinRiese — APPROVED** [review]

**Pull request merged**